youtube_dl/extractor/ceskatelevize.py

```diff
@@ -277,3 +299,34 @@ def _real_extract(self, url):
             webpage, 'iframe player url', group='url'))
 
         return self.url_result(data_url, ie=CeskaTelevizeIE.ie_key())
+
+
+class CeskaTelevizeDeckoIE(InfoExtractor):
+    _VALID_URL = r'https?://decko.ceskatelevize.cz/(?P<id>[a-z-]+)$'
+    _TEST = {
+        'url': 'http://decko.ceskatelevize.cz/nejmensi-slon-na-svete',
+        'playlist_count': 13
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        idec = self._html_search_regex(r'var\s+IDEC\s+=\s+\'(.+?)\'', webpage, 'IDEC')
+
+        args = compat_urllib_parse_urlencode({"IDEC":idec})
+        url = "http://decko.ceskatelevize.cz/rest/Programme/relatedVideosForEpisode?" + args
```

Comment:
Would you please elaborate? I am not sure what exactly do you mean. What method call? Are you suggesting evading these local variables, creating a large `_download_json` call instead? As in,

```python
self._download_json(
  'http://decko.ceskatelevize.cz/rest/Programme/relatedVideosForEpisode?' 
    + compat_urllib_parse_urlencode({'IDEC':idec}),
  video_id
)
```
?